QueryCiv3/Biq.cs

```diff
@@ -0,0 +1,549 @@
+using System;
+using System.Collections.Generic;
+using QueryCiv3.Biq;
+
+namespace QueryCiv3
+{
+    public class BicData
+    {
+        public Civ3File Bic;
+        public bool HasCustomRules => Bic.SectionExists("BLDG");
+        public bool HasCustomMap => Bic.SectionExists("WCHR");
+
+        // RULE, WCHR, WMAP, and GAME all seem to only ever have a maximum of 1 header in the biq files, but I'm not certain on that
+        // If that's confirmed to be the case, they can be demoted from arrays to singular variables and the code simplified accordingly
+        public BLDG[] Bldg;
+        public CITY[] City;
+        public CLNY[] Clny;
+        public CONT[] Cont;
+        public CTZN[] Ctzn;
+        public CULT[] Cult;
+        public DIFF[] Diff;
+        public ERAS[] Eras;
+        public ESPN[] Espn;
+        public EXPR[] Expr;
+        public FLAV[] Flav;
+        public GAME[] Game;
+        public GOOD[] Good;
+        public GOVT[] Govt;
+        public LEAD[] Lead;
+        public PRTO[] Prto;
+        public RACE[] Race;
+        public RULE[] Rule;
+        public SLOC[] Sloc;
+        public TECH[] Tech;
+        public TERR[] Terr;
+        public TFRM[] Tfrm;
+        public TILE[] Tile;
+        public UNIT[] Unit;
+        public WCHR[] Wchr;
+        public WMAP[] Wmap;
+        public WSIZ[] Wsiz;
+
+        /*
+            Note which of the following are rectangular 2d arrays [,] vs which are jagged 2d arrays [][]
+            A rectangular array means that the second dimension for all components is the same eg. for RaceEra, every civ shares the same # of Eras
+            A jagged array means that the second dimension can vary between components eg. for RaceCityName, each civ can have a different # of Cities
+        */
+        public bool[,] TerrGood; // which resources are allowed on which types of terrain
+        public GOVTGOVT[,] GovtGovt; // relationships between governments
+        public RACECITYNAME[][] RaceCityName; // the list of city names for each civ
+        public RACEERA[,] RaceEra; // the file names for each era for each civ
+        public RACELEADERNAME[][] RaceGreatLeaderName; // the great leaders for each civ
+        public RACELEADERNAME[][] RaceScientificLeaderName; // the scientific leaders for each civ
+        public int[][] CityBuilding; // Building IDs in each city
+        public int[][] WmapResource;
+        public int[][] PrtoPrto; // Stealth unit targets per unit
+        public LEADPRTO[][] LeadPrto; // starting unit data for each leader
+        public int[][] LeadTech; // starting tech data for each leader
+        public RULECULT[][] RuleCult; // culture level names per rule
+        public int[][] RuleSpaceship; // spaceship quantity requirements per rule
+        public int[][] GameCiv; // Playable civs for game
+        public int[][] GameAlliance; // Civ alliances for game
```

Comment:
Yeah, having it be ALLUPPERCASE was just a convention I was copying over from the Civ 3 headers in a way that didn't really make sense. But now I've reworked it in a way I think is more clear. I've added underscores, and now only have full uppercase in places where lengths match.

eg. RULECULT becomes RULE_CULT because the outer index of RULE_CULT has the same length as RULE and the inner index has the same length as CULT. However, RACECITY becomes RACE_City because while the outer index matches the length of RACE, the inner index in this context does not match that of CITY. I think this is a pretty good and slightly more clear setup, but let me know what you think.

Reply:
Oh, and I didn't change the pointer names to match the new names of the structs, but I'll do that in the future Biq cleanup PR.

Reply:
That looks much better!  Easy to read at a glance now.  The variable names are probably okay as-is, and in some cases (military/scientific leaders) there are more variables than structs anyway.